pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/AutoProduceValidatedAvroBytesSchema.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.client.impl.schema;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.client.api.schema.SchemaDefinition;
+import org.apache.pulsar.client.internal.DefaultImplementation;
+import org.apache.pulsar.common.schema.SchemaInfo;
+import org.apache.pulsar.common.schema.SchemaType;
+
+import java.util.Optional;
+
+/**
+ * Auto detect schema.
+ */
+public class AutoProduceValidatedAvroBytesSchema<T> extends AutoProduceBytesSchema<T> {
+
+    private Schema<T> schema;
+
+    
+    public AutoProduceValidatedAvroBytesSchema(org.apache.avro.Schema schema) {
+        SchemaDefinition schemaDefinition = SchemaDefinition.builder().withJsonDef(schema.toString(false)).build();
+        
+        setSchema(DefaultImplementation.newAvroSchema(schemaDefinition));
+    }
+
+    public AutoProduceValidatedAvroBytesSchema(Schema<T> schema) {
+        this.schema = null;
```

Comment:
why do you need to set this to `null` if you are setting it in the line below ?

Reply:
I don't know. the early build failed with a bug report about "unread field" or something, which I don't understand at all. 

I thought `setSchema` could only be called on initialized values. In the constructor, it has to initialize all its fields before calling other methods.

This looks "stupid" to me as well. But I don't have other ideas fixing that "bug".